pom.xml

```diff
@@ -510,6 +521,137 @@
           </execution>
         </executions>
       </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-compiler-plugin</artifactId>
+        <version>3.1</version>
+        <configuration>
+          <source>1.8</source>
+          <target>1.8</target>
+        </configuration>
+      </plugin>
+
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-surefire-plugin</artifactId>
+        <version>2.14.1</version>
+        <configuration>
+          <argLine>-Xmx512m</argLine>
+          <systemPropertyVariables>
+            <java.io.tmpdir>${project.build.directory}</java.io.tmpdir>
+          </systemPropertyVariables>
+          <includes>
+            <include>**/*TestSuite.java</include>
+            <include>**/Test*.java</include>
+            <include>**/*Test.java</include>
+            <include>**/*TestCase.java</include>
+          </includes>
+        </configuration>
+      </plugin>
+
     </plugins>
   </build>
+
+  <profiles>
+    <profile>
+      <id>e2e-tests</id>
+      <properties>
+        <testSourceLocation>src/e2e-test/java</testSourceLocation>
+      </properties>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-surefire-plugin</artifactId>
+            <version>2.18.1</version>
+            <configuration>
+              <skipTests>true</skipTests>
+            </configuration>
+          </plugin>
+
+          <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-failsafe-plugin</artifactId>
+            <version>3.0.0-M5</version>
+            <configuration>
+              <includes>
+                <include>TestRunner.java, TestRunnerMand.java, TestRunnerOpt.java</include>
```

Comment:
Use pattern instead of listing all classes

Reply:
@tivv since we have to maintain the execution order of the TestRunners, we had to add the runner filenames like this. We have to make sure TestRunnerOpt runs in the end, else it will impact TestRunnerMand

Reply:
Why do we need an execution order? It sounds like an issue. All tests must be independent. One should be able to run any test separately or tests in any sequence. That's basic principle.

Reply:
@tivv For config driven scenarios, wherein we are implementing cucumberJvm plugin, we are loading optional properties on the run, once they are loaded, the scenarios with only mandatory fields will not run. So mandatory field scenario should run first, then plugin loading the optional properties should execute followed by optional field scenarios. Suggest if I am not able to put forward my point, or if I am missing something to make the scenarios more independent

Reply:
Sure, could you please clarify in more details. Overall approach is that tests must be self-contained and do any necessary clean-up as needed. Why is it not possible here?

Reply:
done, thanks